src/odb/src/db/dbBTerm.cpp

```diff
@@ -1043,11 +1043,13 @@ bool dbBTerm::isMirrored()
   return bterm->_is_mirrored;
 }
 
-bool dbBTerm::isValid() const
+bool dbBTerm::isValid(const dbBTerm* bterm, const dbBlock* block)
 {
-  const _dbBTerm* bterm = (const _dbBTerm*) this;
-  const _dbBlock* block = (const _dbBlock*) bterm->getOwner();
-  return block->_bterm_tbl->validId(bterm->getOID());
+  if (bterm == nullptr || block == nullptr) {
+    return false;
+  }
+  const _dbBlock* block_impl = (const _dbBlock*) block;
+  return block_impl->_bterm_tbl->validId(bterm->getId());
```

Comment:
It would be simpler to just check DB_ALLOC_BIT on the id and not need to look at the table at all.   See
```
void dbTable<T, page_size>::destroy(T* t)
{
...
  o->_oid = oid & ~DB_ALLOC_BIT;
```